### PR TITLE
Refactor frontend

### PR DESCRIPTION
**Pull request by Xillez**

### CODE REVIEW

File: frontend/js/3DView/3DView.js [hunk 1]
```diff
@@ -65,68 +77,157 @@ function render()
...
+
+    // Draw nodes using display manager.
+    fdg.getNodes().forEach((node) => {
+        /**
+         * Function for selecting proper type from configuration.
+         * @param {string} type - The type of node.
+         */
+        let getDrawableGeometry = (function(type) {
+            if (type == "cube")
+            return new THREE.BoxGeometry(0.1, 0.1, 0.1);
```
Comment (zohaib194) [inline frontend/js/3DView/3DView.js]: indent missing here.

Comment (zohaib194) [reply frontend/js/3DView/3DView.js]: Add scopes on if and else if

File: frontend/js/DataProcessing/FileProcessing.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    // Handle any children.
+    handleCodeData(classData.Class);
+
+    // We're done in this part of the tree.
+    indexStack.pop();
+}
+
+/**
+ * Function for handling namepspace data in JSONObject.
+ * @param {*} namespaceData
```
Comment (eldarht) [inline frontend/js/DataProcessing/FileProcessing.js]: *?

File: frontend/js/repository.js [hunk 3]
```diff
@@ -5,21 +5,43 @@ function submitRepoName(){
...
+    }).catch((error) => {
+        console.log("Error: " + error);
+    });
+
 	// Create a http request
-	var xhr = new XMLHttpRequest();
-	var body = 	{
-					'uri': repoName.value
-				};
+	/*var xhr = new XMLHttpRequest();
```
Comment (eldarht) [inline frontend/js/repository.js]: Commented out code? if so remove.

**Pull request merged**